sdc/tests/test_indexes.py

```diff
@@ -0,0 +1,386 @@
+# -*- coding: utf-8 -*-
+# *****************************************************************************
+# Copyright (c) 2020, Intel Corporation All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#
+#     Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+import numpy as np
+import pandas as pd
+import unittest
+
+from itertools import (combinations_with_replacement, product, filterfalse)
+
+from sdc.tests.test_base import TestCase
+from sdc.utilities.sdc_typing_utils import kwsparams2list
+from sdc.tests.test_series import _make_func_from_text
+from numba.errors import TypingError
+
+
+test_global_index_names = [None, 'abc', 'index']
+test_global_range_member_values = [1, 2, 10, -5, 0, None]
+
+
+def _generate_valid_range_params():
+
+    def valid_params_predicate(range_params):
+        # if step is zero or all start/stop/step are None range is invalid
+        return (range_params[-1] == 0
+                or all(map(lambda x: x is None, range_params)))
+
+    return filterfalse(
+        valid_params_predicate,
+        combinations_with_replacement(test_global_range_member_values, 3)
+    )
+
+
+class TestRangeIndex(TestCase):
+
+    def test_range_index_create_and_box(self):
+        def test_impl(start, stop, step, name):
+            return pd.RangeIndex(start, stop, step, name=name)
+        sdc_func = self.jit(test_impl)
+
+        for params in _generate_valid_range_params():
+            for name in test_global_index_names:
+                start, stop, step = params
```

Comment:
This line could be moved from internal loop to external one. The action could be applied in several places.

Reply:
Yes, thank you, fixed it.